lib/screens/app/wallet/scan.dart

```diff
@@ -29,23 +29,44 @@ class _ScanState extends State<Scan> {
     setState(() {
       this.step = Steps.scan;
     });
-
+    //return;
     try {
-      ScanResult scanResult = await BarcodeScanner.scan();
-      if (scanResult.type == ResultType.Cancelled) {
-         Navigator.of(context).pop();
-         
-      } else {
-        setState(() {
-          this.step = Steps.processing;
-          this.qrcode = scanResult.rawContent;
-        });
-        processSigningRequest();
+      bool shouldKeepScanning = true;
+      while (shouldKeepScanning) {
+        ScanResult scanResult = await BarcodeScanner.scan();
+        if (scanResult.type == ResultType.Cancelled) {
+          shouldKeepScanning = false;
+          Navigator.of(context).pop();
+          break;
+        } else {
+          setState(() {
+            this.step = Steps.processing;
+            this.qrcode = scanResult.rawContent;
+          });
+
+          print("QR Code: " + scanResult.rawContent);
+          var esr;
+          try {
+            esr = await EosioSigningRequest.factory(
+              EosService.of(context, listen: false).client,
+              scanResult.rawContent,
+              SettingsNotifier.of(context, listen: false).accountName,
+            );
+          } catch (e) {
+            print("can't parse ESR " + e.toString());
+            print("ignoring...");
+          }
+          if (esr != null && canProcess(esr)) {
+            shouldKeepScanning = false;
+            processSigningRequest(esr);
+            break;
+          }
+        }
       }
     } on PlatformException catch (e) {
       if (e.code == BarcodeScanner.cameraAccessDenied) {
         setState(() {
-          this.error = 'The user did not grant the camera permission!';
+          this.error = 'Please enable camera access to scan QR codes!';
```

Comment:
is this error displayable to the user? Does it need translation ?

Reply:
Yes it does need translation

Reply:
Google Habilite el acceso a la cámara para escanear el código QR